### PR TITLE
Auto create animation

### LINKED ISSUE #96: Auto create AnimatedDrawable from ImageDrawable
- Consider the image drawable as a tilesheet with the frames
- Open a popup to collect the info about frame size, gap size between frames, etc

### PR DESCRIPTION
**Pull request by tlein**
Resolves #96

### CODE REVIEW

**Comment by tlein** [comment]
Why does the contents of the dialog box look blurry?

**Comment by tlein** [comment]
Should we account for incorrect dimensions, the tiles being on more than one row, uniform gaps between frames?

**Comment by jeffswenson** [comment]
Future Issues
This doesn't do any validation on the image.
There isn't a way to set an offset.

File: src/duckling/controls/section-header.component.ts [hunk 1]
```diff
@@ -54,4 +57,12 @@ export class SectionHeaderComponent {
             this.sectionOpenChanged.emit(sectionOpened);
         }
     }
+
+    get headerCssClasses() : string {
+        let classes = ["header", "md-elevation-z3"];
```
Comment (jeffswenson) [inline src/duckling/controls/section-header.component.ts]: You should replace this with
`
if (this.collapsible) {
    return "header md-elevation-z3";
}
return "header md-elevation-z3 pointer";
`

File: src/duckling/game/drawable/auto-create-animation-dialog.component.ts [hunk 2]
```diff
@@ -8,43 +8,113 @@ import {
...
+    onDimensionInput(newFrameDimensions : Vector) {
+        this.frameDimensions = newFrameDimensions;
+    }
+
+    onNumFramesInput(newNumFrames : number) {
+        this.numFrames = newNumFrames;
+    }
+
+    onImageFilePicked(newImageKey : string) {
+        let homeResourceString = this._path.join(this._project.project.home, 'resources');
```
Comment (jeffswenson) [inline src/duckling/game/drawable/auto-create-animation-dialog.component.ts]: This is a direct copy of a method in image-drawable.component. You should extract it.

Comment (tlein) [reply src/duckling/game/drawable/auto-create-animation-dialog.component.ts]: fixed

File: src/duckling/controls/section-header.component.ts (hunk 1 above)
Comment (tlein) [reply src/duckling/controls/section-header.component.ts]: fixed

File: src/duckling/controls/browse-asset.component.ts [hunk 3]
```diff
@@ -41,10 +42,15 @@ export class BrowseFileComponent {
     @Input()
     selectedFile = "";
 
+    @Input()
+    postFileChoiceHook : (file : string) => boolean;
```
Comment (jeffswenson) [inline src/duckling/controls/browse-asset.component.ts]: This is a sexy hook, but you aren't using it anywhere.

**Pull request merged**